viewing_party/party.py

```diff
@@ -1,23 +1,175 @@
+import pytest
+
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    
+    if title and genre and rating:
+            return {
+                "title": title,
+                "genre": genre,
+                "rating": rating
+            }
+    return None
+
+
+def add_to_watched(user_data, movie):
+    user_data['watched'].append(movie)  
+    return user_data
+    
+
+def add_to_watchlist(user_data, movie):
+    user_data['watchlist'].append(movie)
+    return user_data
+
+
+def watch_movie(user_data,title):
+    movie_to_move = None
+    for movie in user_data['watchlist']:
+        if movie['title'] == title:
+            movie_to_move = movie
+            break
+          
+    if movie_to_move:
+        user_data['watchlist'].remove(movie_to_move)
+        user_data['watched'].append(movie_to_move)
+    
+    return user_data

```

Comment:
Great work! Love how you used the `break` keyword here as well since once you find a movie you don't need to look through the rest of it! Here I want to talk about the `remove` method. Remember that `remove` has an `O(n)` time complexity. There are a couple of ways that we could circumvent this. Here is one way that doesn't change your code as much: 

```python
def watch_movie(user_data,title):
    index = None
    watchlist = user_data["watchlist"]

    for i in range(len(watchlist)):
        movie = watchlist[i]

        if movie['title'] == title:
            index = i
            break
          
    if index is not None:
        movie_to_move = watchlist.pop(index)
        add_to_watched(user_data, movie_to_move)
    
    return user_data
   ```
   Also in this implementation, we will only use `remove` once, but in other filtering problems we might have to do this for multiple elements so it doesn't hurt to get in the habit now.